src/server/worker/pipeline/spout/spout.go

```diff
@@ -7,6 +7,13 @@ import (
 
 // Run will run a spout pipeline until the driver is canceled.
 func Run(driver driver.Driver, logger logs.TaggedLogger) error {
+	// If the output branch head commit is open, close it
```

Comment:
Why are we doing this? The user should manage the state of the branches after a restart (they have full access to the repo through the pachclient with spouts 2.0). This would impose semantics on them that may not line up with their use case. Also, the branches should have semantics similar to source branches (I've started using this term rather than input because output branches can be inputs for downstream pipelines) rather than the output branches of transform pipelines.

Reply:
Since branches now have default head commits, PFS generates an open commit for the spout's output branch.  There might be other ways to solve this, but PFS has no idea if the output branch is for a spout or other type of pipeline.  Open to suggestions.

Reply:
It looks like you already made this change, but what makes sense to me is for the empty commit to be finished at creation time. It looks like you made a change to finish the empty commit when there is no provenance or when the only provenance is a branch in the spec repo.